kube-core/src/labels.rs

```diff
@@ -183,9 +148,9 @@ impl Selector {
     }
 }
 
-impl Matcher for Selector {
+impl Matcher<Map> for Selector {
```

Comment:
by putting the generic here you can only do one impl per map type, you could instead do:

```
impl Matcher for Selector {
   type Map = BTreeMap<String, String>
```
with an associated type and reference `Self::Map` rather than `Map` in signatures

Reply:
I feel like at this point it can be anything. Left it on `type Items` but was also thinking about `type Item`?

Reply:
yeah, which i think is a good thing for future versatility; clarified in https://github.com/kube-rs/kube/pull/1539#discussion_r1684459281